examples/AspNetCore/Program.cs

```diff
@@ -13,16 +13,16 @@
 var appBuilder = WebApplication.CreateBuilder(args);
 
 // Note: Switch between Zipkin/OTLP/Console by setting UseTracingExporter in appsettings.json.
-var tracingExporter = appBuilder.Configuration.GetValue("UseTracingExporter", defaultValue: "console")!.ToLowerInvariant();
```

Comment:
Hi @SimonCropp, if you can include your justification for why these are incorrect, that will make the code review much easier. Otherwise our reviews need to do more work to individually verify each of these statements.

For example; This looks safe to remove because by supplying a defaultValue, this method will never return null.

Reply:
the IDE highlights them as redudnant, since it know the nullability in each case, and i double checked each of them. are there any you think need more justification

Reply:
What IDE or plugin are you using? I'm not seeing the same highlights in Visual Studio but I would like to.

Reply:
@TimothyMothra Rider. but you would get the same using R#

Reply:
Is it giving you any code names for these items? something like CS### or IDE### ?
I'm curious if this is something we could turn on in our CI.

Reply:
friendly ping. Is your tooling providing any codes that we could include in our [editorconfig](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/.editorconfig)?
This would be the best way to codify your changes so we don't have regressions in the future.

Reply:
it is a jetbrains only setting https://www.jetbrains.com/help/rider/RedundantSuppressNullableWarningExpression.html

Reply:
@TimothyMothra, there is a chance that also `AnalysisLevel` correctly set could find such issues. Not tested: https://github.com/open-telemetry/opentelemetry-dotnet/issues/3958